### ISSUE TITLE
Tweaking of permission levels

### ISSUE THREAD

**Issue by MinoMino** [MinoMino/minqlx#8]
Permission levels are very loosely defined. They should have a more clear definition of what the various levels should and should not allow.

**Comment by mattiZed** [comment]
Maybe these levels should be somewhat based on what they can do to the game. On my minqlbots i had them kinda set up like this:

Permlevel 1 -> Pass certain restrictions
Permlevel 2 & 3 -> Match Management
Permlevel 4 -> User Management
Permlevel 5 -> Owner

Let me explain:
- Permlevel 1
  I used that as a barrier breaker. When someone was restricted to the game because of elo-limits or not enough games_completed I could grant them Level 1 to pass these restrictions. Was always handy to have that - not much of a use now, but lateron?
- Permlevel 2
  Here I grouped anything that would have to do with "light" match managing. Passing and vetoing votes, shuffling, putting players around, balancing, allready, abort, etc.
- Permlevel 3
  Here I grouped anything that would have to do with "heavy" match managing. Things as kicking, kickbanning, permanently banning and mapchanging via !map (i consider this a major match management when not changing the map via democratic voting)
- Permlevel 4
  Users that had Permlevel 4 could promote and demote users, but only up to their own level. No one but the Server Owner could set Permission Level 5 for users.
- Permlevel 5
  Level of the serverowner. I always kept this to myself.

**Comment by MinoMino** [comment]
Thank you for the input. I think it'd be cleaner to instead implement some sort of way to exempt players from restrictions instead of using perm levels, but the idea of using level 1 as a non-administrative level is interesting and something I will consider. The rest sounds solid.

**Comment by mattiZed** [comment]
Yeah, thats just the way I handled it. Maybe gives some thoughts!

**Comment by pRo-b1ngo** [comment]
I like that system from mattized. i have one suggestion for the kick and ban feature. i would like to have kick in perm level 2 and ban in 3.
